CRM/Contribute/Form/ManagePremiums.php

```diff
@@ -291,10 +292,16 @@ public function postProcess() {
     $this->_processImages($params);
 
     // Save the premium product to database
-    $premium = CRM_Contribute_BAO_Product::create($params);
+    // If we're updating, we need to pass in the premium product Id
```

Comment:
Actually you don't. `APIv4::update` doesn't require a `where` param if `$values` contains an `id` :)